pdc/apps/unreleasedvariant/views.py

```diff
@@ -0,0 +1,205 @@
+#
+# Copyright (c) 2018 Red Hat
+# Licensed under The MIT License (MIT)
+# http://opensource.org/licenses/MIT
+#
+from pdc.apps.common import viewsets
+from pdc.apps.module.models import Module
+from pdc.apps.unreleasedvariant.serializers import UnreleasedVariantSerializer
+from pdc.apps.unreleasedvariant.filters import UnreleasedVariantFilter
+
+
+class UnreleasedVariantViewSet(viewsets.PDCModelViewSet):
+    """
+    ##Overview##
+
+    This page shows the usage of the **Module API**, please see the following for more details.
+    """
+    model = Module
+    queryset = Module.objects.all().order_by('uid')
+    filter_class = UnreleasedVariantFilter
+    serializer_class = UnreleasedVariantSerializer
+    lookup_field = 'uid'
+
+    def list(self, request, *args, **kwargs):
+        """
+        __Method__:
+        GET
+
+        __URL__: $LINK:unreleasedvariants-list$
+
+        __Query Params__:
+
+        %(FILTERS)s
+
+        __Response__:
+
+            {
+                "count": int,
+                "next": url,
+                "previous": url,
+                "results": [
+                    {
+                        "variant_id": string,
+                        "variant_uid": string,
+                        "variant_name": string,
+                        "variant_type": string,
+                        "variant_version": string,
+                        "variant_release": string,
+                        "variant_context": string,
+                        "koji_tag": string,
+                        "modulemd": string,
+                        "active": bool,
+                        "runtime_deps": [{"dependency": string, "stream": string}, ... ],
+                        "build_deps": [{"dependency": string, "stream": string}, ... ],
+                        "rpms": [string, ... ]
+                    },
+                    ...
+                ]
+            }
+        """
```

Comment:
You can remove these docstrings if the generated API documentation looks OK.

It would be difficult to keep these up-to-date in future (nobody updates comments).

Reply:
Sorry, just realized that it's still not possible to remove these docstrings. Would be at least nice to use `%(SERIALIZER)s` where possible.

Reply:
Good idea. I had forgotten PDC could do this.